### ISSUE TITLE
f7 functions not working anymore

### ISSUE THREAD

**Issue by Kanaha150** [RinteRface/shinyMobile#235]
I have a PWA created based on R using shiny, shinyMobile, shinyWidgests packages, among others. The app was functioning fine until last week (mid-Nov 2022) when I commented out a few texts for the first time since April 2022. All the reactive components of the app don't show anymore, even when running app locally. I reinstalled all the shiny related packages, uncommented the texts back (and using the exact script from April), and updated R and RStudio, but still not working. I do not have any errors in R console. But I see errors in the web console when I run the app locally (see the photo attached). I was thinking it is to be with updates between April and Nov this year, but cannot figure out which one (R, R package, f7framework or something else). It could also be something else. Any suggestion or advice here? Thanks in advance!

![image](https://user-images.githubusercontent.com/71273013/203076658-2d786662-0e0c-42b8-bd7e-954f4503d88e.png)

**Comment by DivadNojnarg** [comment]
Did you try shinyMobile rc1.0.0 branch? Recent shiny versions have a breaking change on the JavaScript side causing this kind of issue. This branch was also submitted to CRAN about 2 weeks ago and I am still waiting for the acceptance. In the meantime you can install from GitHub (not from master but from rc-1.0.0)

**Comment by DivadNojnarg** [comment]
Also please look at this change list in case you update: 

https://github.com/RinteRface/shinyMobile/blob/rc-1.0.0/NEWS.md#breaking-change

**Comment by Kanaha150** [comment]
Thanks a lot for your reply!
I just tried that and then this error appeared (which was not the case back in April 2022...)
![image](https://user-images.githubusercontent.com/71273013/203136796-36e97ab0-8be4-4dc8-935e-70e94d0609cc.png)
So I removed" "(space) I had for some tabname and run again. and all the reactive plots etc came back!
However, the theme color selection still does not work (only changes the color in the panel but not the theme and it is always dark theme) (see the photo below)
![image](https://user-images.githubusercontent.com/71273013/203137901-def37125-d361-436d-8198-0dfaf2c8ecd6.png)
and also I don't see the name of the tabs anymore... I see that in the list you shared, Tabs will have to be manually renamed. How can I do that? for now, it looks like this and I only see the icon.
![image](https://user-images.githubusercontent.com/71273013/203138741-6b5dd178-2ee9-416a-a86b-4395e8c66f64.png)
Thanks!

**Comment by DivadNojnarg** [comment]
For the tabs you have to pass an extra title parameter with the tab title. Could you show me the app code?

**Comment by Kanaha150** [comment]
For a record, I found a way! I added
`$("div").removeClass("view view-main theme-dark");` after
`$("html").removeClass("theme-dark");` and that worked:)
For others, after figuring that out, there were some errors when publishing the app because "charpente" package was not installed. I could not install it first because some dependencies (incl. "erratum") were not available. So either I downloaded it directly from GitHub if possible. For "erratum", it was achieved, so I downloaded it from the archived folder... It might have been better to modify the app code itself to adapt to the latest packages, but in my case, I just wanted to make the app up on running again and have no plan in the future to modify or upgrade.
Anyhow, big thanks to David for creating all these packages! It was really thanks to them that I was able to make an app in the first place.
